node/actors/network/src/gossip/attestation_status.rs

```diff
@@ -0,0 +1,85 @@
+use std::sync::Arc;
+
+use zksync_concurrency::{ctx, sync};
+use zksync_consensus_roles::attester;
+use zksync_consensus_storage::BatchStore;
+
+use crate::watch::Watch;
+
+/// An interface which is used by attesters and nodes collecting votes over gossip to determine
+/// which is the next batch they are all supposed to be voting on, according to the main node.
+#[async_trait::async_trait]
+pub trait AttestationStatusClient: 'static + std::fmt::Debug + Send + Sync {
+    /// Get the next batch number for which the main node expects a batch QC to be formed.
+    ///
+    /// The API might return an error while genesis is being created, which we represent with `None`
+    /// here and mean that we'll have to try again later.
+    async fn next_batch_to_attest(
+        &self,
+        ctx: &ctx::Ctx,
+    ) -> ctx::Result<Option<attester::BatchNumber>>;
```

Comment:
How about we shift the abstraction a bit and make the AttestationReceiver abstract? Sth like/equivalent to
```
trait AttestationReceiver {
  fn subscribe() -> watch::Receiver<Option<attester::BatchNumber>>
}
```
I would like to push the active polling as far out as possible, to avoid hardcoded delays which the outer logic has to be aware of. In this case the polling still will be performed, but in zksync-era code.

Reply:
You mentioned before that you might want to extend the `AttestationStatus` with further things like the number of concurrent votes, which I expected would not change during the lifetime of the program, but may be announced by the main node. I suppose we can add further async methods to `AttestationReceiver` above that one can call.

Reply:
or we can have one large watch, it should work either way

Reply:
technically we could even use the watch `instead` of the `AttestationReceiver` trait, except that it would limit extensionality of the API. But I guess we can revert to having a trait at any point.

Reply:
I do wonder though if we should just expose the `AttestationStatusWatch` itself as a parameter to the `Explorer`. There isn't much magic there, it has `update` and `subscribe`, which can be driven from `zksync-era`.

Reply:
yes, and imo a piece of shared state with a well designed API is more maintainable API surface than a trait interface.

Reply:
I agree with encapsulating things and having shared memory be a good integration pattern. I have a Software Transactional Memory library that could be adopted to work with ctx to make this composable as well.

Reply:
I am a bit worried though that moving the poll onto the era side will make it more cumbersome to compose the application now. The Network is now self contained so the vote propagation tests did not have to change. If it just sits there doing nothing, and not even the Executor has the stuff we just make it more fragmented. I’ll do it and we’ll see. After all the PermanentStore variants do it too, maybe this is a general component to watch the main node.

Reply:
Created an `AttestationStatusRunner` to encapsulate the polling, created by `let (status, runner) = AttestationStatusWatch::new(client, poll_interval);`, which should be done where we instantiate the `BlockStore`, the `BatchStore` and their respective runners. This will need to be duplicated for the main and external node setups, and then only the `AttestationStatusWatch` is passed to `Executor` and then onto the `Network`.

Reply:
We shouldn't need to define the AttestationStatusClient now, no?

Reply:
like, AttestationStatusWatch is our API, and how it will be fed information is up to the caller.

Reply:
It's for convenience really, to be used by the `AttestationStatusRunner` so you don't have to write that loop, just hook in either the database or the API based query method. If one doesn't want to use the runner, they don't have to.

Reply:
Moved.

Reply:
thanks